redash/models.py

```diff
@@ -1000,6 +1006,32 @@ def outdated_queries(cls):
 
         return outdated_queries.values()
 
+    @classmethod
+    def delete_stale_resultsets(cls):
+        delete_count = 0
+        queries = Query.query.filter(Query.schedule_resultset_size != None).order_by(Query.schedule_resultset_size.desc())
+        # Multiple queries with the same text may request multiple result sets
```

Comment:
I assume when you say `queries with the same text` you mean same SQL?

Reply:
Yes. (Though of course this is used for non-SQL data sources as well.)